src/pages/index.js

```diff
@@ -2,76 +2,215 @@ import "./index.css";
+);
+const deleteFormButton = document.querySelector(".cards__delete-button");
 
 const cardTemplateSelector = "#card-template";
 
-const editCardModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-editCardModal.setEventListeners();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "6a75e942-1cdf-4570-be93-f1316f67106a",
+});
 
-const previewImageModal = new PopupWithImage("#preview-image-modal");
-previewImageModal.setEventListeners();
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const itemSection = new Section({
-  items: initialCards,
-  renderer: renderCard,
-  cssSelector: ".cards__list",
+let itemSection;
+
+api.getCardList().then((items) => {
+  itemSection = new Section({
+    items,
+    renderer: renderCard,
+    cssSelector: ".cards__list",
+  });
+
+  itemSection.renderItems(items);
 });
-itemSection.renderItems();
 
-const addCardModal = new PopupWithForm(
-  "#profile-add-card",
-  handleCardAddSubmit
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-addCardModal.setEventListeners();
 
-function handleImageClick(name, link) {
-  previewImageModal.open(name, link);
+function renderCard(card) {
+  const cardElement = createCard(card);
+  itemSection.addItem(cardElement);
 }
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    // a = true
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      console.log(card);
+      card.remove();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+    deleteConfirm.close();
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
+function handleLikeClick(card) {
+  api
+    .toggleCardLike(card._id, card.isLiked)
+    .then((_isLiked) => {
+      // console.log("RESPONSE");
+      // console.log(_isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleChangePicture(avatar) {
+  changePictureModal.setEventListeners();
+  console.log(avatar);
+  this._buttonElement = this._popupForm.querySelector(".modal__button");
+  this._buttonElement;
+  changePictureModal.renderLoading(true);
+  api
+    .setUserAvatar(avatar)
+    .then((data) => {
+      changePictureModal.renderLoading(false);
```

Comment:
`      changePictureModal.renderLoading(false);
`
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.